### ISSUE TITLE
When contact is permanently deleted, sdd mandate is still kept. Should this not also be deleted?

### ISSUE THREAD

**Issue by ErikHommel** [Project60/org.project60.sepa#290]

**Comment by systopia** [comment]
I would say: NO.

If you delete a mandate (which contains information that you usually don't want to lose), you have to decide what to do with the contributions. And the transaction groups: if you take contributions out, they become inconsistent... and so on.

But of course I'm open to a discussion.

**Comment by tttp** [comment]
@systopia good point, but if you delete the contact, the contributions are deleted anyway, aren't they? and so it becomes inconsistent anyway, isn't it?

(not saying it's a good thing, just that keeping the mandate isn't going to save us)

**Comment by systopia** [comment]
Not sure... deletion might be blocked - for exactly that reason. Maybe someone could test this...?

**Comment by ErikHommel** [comment]
Should then probably be the other way around: contact should be permanently deleted if it still has sdd mandates? The same is true for (recurring) contributions.

**Comment by systopia** [comment]
> contact should be permanently deleted if it still has sdd mandates?

Could you elaborate, I think I don't understand

**Comment by magnolia61** [comment]
Sidenote: It would the probably also be wise to take in to account status 'deceased', although there is a whole other logic behind that workflow.

**Comment by ErikHommel** [comment]
@systopia I mean you should NOT be able to permanently deleted the contact if there are still SDD mandates for the contact?

**Comment by systopia** [comment]
@ErikHommel: Ok, yes, that sounds right. We'd have to implement the hook, I suppose.
@magnolia61: I agree, there might be a complex workflow behind this (deceased, 'in trash', etc..). I think this is beyond the scope of CiviSEPA, but if we can do anything to support those workflows in a generic way....

**Comment by mielert** [comment]
I can't imagine how a mandate should be still valid when a contact is deleted or deceased. At least the reference has to be changed if i.e. a mandate moves from household to individual after one person in household has deceased.
For me, blocking of deletion is no solution. The contact is much more important than the mandate. The mandate has to be deleted if a contact is deleted permanently. If that's not possible, we've got to ignore mandates of deceased or deleted contacts.
If you do not agree: In which case could a mandate of a deceased or deleted contact still be valid?

**Comment by systopia** [comment]
First I would like to take the "contact deceased" debate out the equation here, @mielert please open a new ticket for this, if you're still interested.

Then, I see the following options for deletion (not just trash can):
 1. **Cascade**, i.e. delete all mandates and contributions. This will lead in incomplete SEPA and contribution records as well as to corrupted SEPA groups
 2. **Detach**, i.e. allow the mandates to live without a contact association. This would probably require some code changes, and it doesn't resolve the situation for the contributions and recurring contributions. AFAIK you cannot delete the contact unless you deleted all the contributions.
 3. **Archive**, i.e. move all contributions, mandates, recurring contributions to a dummy contact (e.g. "Unknown Contributor"). If you don't want to do this manually, this would be a whole new feature.

What do you think?

**Comment by bjendres** [comment]
One more thought: if we *delete* the mandate record, we have no records of the mandate *reference* any more, and cannot guarantee that it won't be re-used (although most reference generators shouldn't).
